### ISSUE TITLE
AOT compilation error using Angular4

### ISSUE THREAD

**Issue by kadosh1000** [shlomiassaf/angular2-modal#369]
I am using version 3.0.1 of the package with Angular 4.
I am getting the following error when trying to build my app using aot:

![image](https://user-images.githubusercontent.com/2615945/27535953-93eea06a-5a75-11e7-8f60-a2b656b01c00.png)

The problem seems to be with the import of `angular2-modal/plugins/bootstrap`

Have anyone encoutered it before?
Any solutions?

**Comment by stevie6410** [comment]
+1 I was just about to submit an issue on this same subject

**Comment by shlomiassaf** [comment]
Lib and angular versions?

**Comment by kadosh1000** [comment]
@shlomiassaf 
As I've written in the top of my issue.
I am using version 3.0.1 of the lib and angular 4.0.2

**Comment by stevie6410** [comment]
Sorry, I am also on Angular 4 and modal v3.0.1

**Comment by shlomiassaf** [comment]
This looks like a configuration thing...

I guess path transformation of the angular plugin, can you post something to GitHub so I can reproduce ?

**Comment by stevie6410** [comment]
I'll try and extract into an example.

**Comment by fatters** [comment]
@stevie6410 How's the example coming along? Shall I set one up?

**Comment by stevie6410** [comment]
Thanks, that would be great. I have been tied up all day.

**Comment by fatters** [comment]
Well for ease of use for a quick setup I used angular cli (the project I'm getting the error on isn't cli). A bare bones button to open a bootstrap modal appears to build fine. Are you guys having issues with a cli project or are you using ngtools/webpack like me. Might be an issue with that?

[Repo here](https://github.com/fatters/angular2-modal-issue). I don't have time now but tomorrow I'll decouple it from the cli and change the AoT loader to the version my actual project is running.

**Comment by stevie6410** [comment]
Hi, I also just quickly threw together an angular CLI project and it works okay. However my main project is also CLI. 

I just tried turning off AOT on the angular cli for the prod build. And it did then build succesfully

`ng build --prod --aot=false`

This dosent answer the question for AOT compilation failures, but it is now working in my main project with this flag...

**Comment by kadosh1000** [comment]
@fatters I am indeed using webpack.
@stevie6410  it also works in my project when I don't use AOT, is your build succesfull without that flag?

**Comment by stevie6410** [comment]
@kadosh1000 Yes. When I set the --aot=false it builds fine.

**Comment by fatters** [comment]
After ejecting and using the exact same version of ngtools/webpack (1.4.2). I'm getting the same thing so I don't think it's a version issue.

`ERROR in .../web/aot/node_modules/angular2-modal/plugins/bootstrap/src/angular2-modal-bootstrap.ng-flat.ngfactory.ts (10,26): Cannot find module '../../../../../../angular2-modal/plugins/bootstrap'.`

It's still trying to find it in /web/angular2-modal instead of web/node_modules/angular2-modal. I'll carry on investigating.

**Comment by shlomiassaf** [comment]
@fatters I cloned the repository

![image](https://user-images.githubusercontent.com/5377501/27722520-e0560ed6-5d70-11e7-8d19-5e96382f5e73.png)

Everything works great, compiles and also runs in the browser....

**Comment by shlomiassaf** [comment]
Maybe it's a FS mapping thing...

Can you go to the `node_modules` folder: `node_modules/angular2-modal/plugins/bootstrap`

There you will find a `package.json` file

At the bottom you will find the file definition section:

```
  "main": "bundle/angular2-modal-bootstrap.rollup.umd.js",
  "module": "bundle/angular2-modal-bootstrap.es5.js",
  "es2015": "bundle/angular2-modal-bootstrap.js",
  "typings": "src/angular2-modal-bootstrap.ng-flat.d.ts"
```

Add `./` as a prefix to each item

```
  "main": "./bundle/angular2-modal-bootstrap.rollup.umd.js",
  "module": "./bundle/angular2-modal-bootstrap.es5.js",
  "es2015": "./bundle/angular2-modal-bootstrap.js",
  "typings": "./src/angular2-modal-bootstrap.ng-flat.d.ts"
```

long shot but it might surprise.

**Comment by DavidMaza** [comment]
I'm also getting this error and tried editing the paths in `node_modules/angular2-modal/plugins/bootstrap/package.json`

Though I still got the same error message when compiling with AngularCLI using AOT.

Using:
- angular2-modal: 3.0.0
- angular: 4.0.0
- angular/cli: 1.1.1

**Comment by shlomiassaf** [comment]
Guys, I cant help without being able to reproduce...

Please state your OS, maybe we can find common ground here...

**Comment by kadosh1000** [comment]
@shlomiassaf it happens to me on both my build server (linux) and on my pc (windows)

**Comment by DavidMaza** [comment]
I managed to track down what was causing this issue for me.

It was because I was using an older version of Angular (v4.0.0). The error does not appear when using the latest version of Angular.

More specifically, I believe it was fixed in Angular 4.2.0-beta.1.

I can reproduce the error in Angular 4.2.0-beta.0 and below, but not in 4.2.0-beta.1 and above.

The only change in Angular 4.2.0-beta.1 was to "add .ngsummary.ts files to support AOT unit tests".  I believe this is what may have fixed the issue.

https://github.com/angular/angular/commit/547c363
https://github.com/angular/angular/blob/master/CHANGELOG.md
